toolchain/sem_ir/typed_insts.h

```diff
@@ -666,7 +669,9 @@ struct ImportRefLoaded {
 // `src_id`, by performing a final copy from source to destination, for types
 // whose initialization is not in-place.
 struct InitializeFrom {
-  // TODO: Make Parse::NodeId more specific.
+  // Note this Parse::NodeId is unused. InitializeFrom is only constructed by
+  // reusing locations.
+  // TODO: Figure out if there's a better way to handle this case.
```

Comment:
How is this different from all the types that use `InvalidNodeId`, e.g. `Temporary`?

Reply:
I'm not sure what semantic is intended here. See my comments to zygoloid above. But I think it's something we can handle separately from this PR.